x/rewards/keeper/min_cons_fee.go

```diff
@@ -0,0 +1,65 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/archway-network/archway/pkg"
+	"github.com/archway-network/archway/x/rewards/types"
+)
+
+// UpdateMinConsensusFee calculates and updates the minimum consensus fee if eligible emitting an event.
+func (k Keeper) UpdateMinConsensusFee(ctx sdk.Context, inflationRewards sdk.Coin) {
+	// Prepare and verify inputs
+	if inflationRewards.IsZero() {
+		k.Logger(ctx).Info("Minimum consensus fee update skipped: inflation rewards are zero")
+		return
+	}
+	inflationRewardsAmt := inflationRewards.Amount.ToDec()
+
+	blockGasLimit := pkg.NewDecFromUint64(ctx.BlockGasMeter().Limit())
+	if blockGasLimit.IsZero() {
+		k.Logger(ctx).Info("Minimum consensus fee update skipped: block gas limit is not set")
```

Comment:
I think this should yield to a panic, but maybe let's keep it this way for testing. This should never hit in main net though.

Reply:
I've left this one 'cause by default, block gas limit is not set. If someone will run our binary, he can face a panic with the misconfiguration.